src/mesh/curved_mesh.jl

```diff
@@ -11,44 +11,34 @@ can be used as domain faces.
 """
 mutable struct CurvedMesh{NDIMS, RealT<:Real} <: AbstractMesh{NDIMS}
   cells_per_dimension::NTuple{NDIMS, Int}
-  faces::Any # Not relevant for performance
-  faces_as_string::Vector{String}
+  mapping::Any # Not relevant for performance
+  mapping_as_string::String
   periodicity::NTuple{NDIMS, Bool}
   current_filename::String
   unsaved_changes::Bool
 end
 
 
 """
-    CurvedMesh(cells_per_dimension, faces, RealT; unsaved_changes=true, faces_as_string=faces2string(faces))
+    CurvedMesh(cells_per_dimension, mapping, RealT; unsaved_changes=true, mapping_as_string=mapping2string(mapping, length(cells_per_dimension)))
 
 Create a CurvedMesh of the given size and shape that uses `RealT` as coordinate type.
 
 # Arguments
 - `cells_per_dimension::NTupleE{NDIMS, Int}`: the number of cells in each dimension.
-- `faces::NTuple{2*NDIMS, Function}`: a tuple of `2 * NDIMS` functions that describe the faces of the domain.
-                                      Each function must take `NDIMS-1` arguments.
-                                      `faces[1]` describes the face onto which the face in negative x-direction 
-                                      of the unit hypercube is mapped. The face in positive x-direction of
-                                      the unit hypercube will be mapped onto the face described by `faces[2]`.
-                                      `faces[3:4]` describe the faces in positive and negative y-direction respectively 
-                                      (in 2D and 3D).
-                                      `faces[5:6]` describe the faces in positive and negative z-direction respectively
-                                      (in 3D).
+- `mapping::Function`: a function of `NDIMS` variables to describe the mapping, which transforms 
+                       the reference mesh to the physical domain.
```

Comment:
This does not have to be a function, right? It could be a callable struct?
```suggestion
- `mapping`: a function of `NDIMS` variables to describe the mapping, which transforms 
             the reference mesh to the physical domain.
```